pkg/kubelet/cm/internal_container_lifecycle_windows.go

```diff
@@ -20,30 +20,119 @@ limitations under the License.
 package cm
 
 import (
-	"k8s.io/api/core/v1"
+	"fmt"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
 	utilfeature "k8s.io/apiserver/pkg/util/feature"
 	runtimeapi "k8s.io/cri-api/pkg/apis/runtime/v1"
 	"k8s.io/klog/v2"
 	kubefeatures "k8s.io/kubernetes/pkg/features"
 	"k8s.io/kubernetes/pkg/kubelet/winstats"
+	"k8s.io/utils/cpuset"
 )
 
 func (i *internalContainerLifecycleImpl) PreCreateContainer(pod *v1.Pod, container *v1.Container, containerConfig *runtimeapi.ContainerConfig) error {
-	if i.cpuManager != nil && utilfeature.DefaultFeatureGate.Enabled(kubefeatures.WindowsCPUAndMemoryAffinity) {
-		allocatedCPUs := i.cpuManager.GetCPUAffinity(string(pod.UID), container.Name)
-		if !allocatedCPUs.IsEmpty() {
-			var cpuGroupAffinities []*runtimeapi.WindowsCpuGroupAffinity
-			affinities := winstats.CpusToGroupAffinity(allocatedCPUs.List())
-			for _, affinity := range affinities {
-				klog.V(4).InfoS("Setting CPU affinity", "container", container.Name, "pod", pod.Name, "group", affinity.Group, "mask", affinity.MaskString(), "processorIds", affinity.Processors())
-				cpuGroupAffinities = append(cpuGroupAffinities, &runtimeapi.WindowsCpuGroupAffinity{
-					CpuGroup: uint32(affinity.Group),
-					CpuMask:  uint64(affinity.Mask),
-				})
-			}
-
-			containerConfig.Windows.Resources.AffinityCpus = cpuGroupAffinities
+	if !utilfeature.DefaultFeatureGate.Enabled(kubefeatures.WindowsCPUAndMemoryAffinity) {
+		return nil
+	}
+
+	klog.Info("PreCreateContainer for Windows")
```

Comment:
you may want to set this to `V(2)` at least, likely `V(4)` otherwise it can get spammy very fast

Reply:
updated to v(4)